tests/src/Simple/Interfaces/Interfaces.cs

```diff
@@ -234,4 +237,52 @@ public string GetAString()
         }
     }
     #endregion
+
+    #region Array Interfaces Test
+    private static int TestArrayInterfaces()
+    {
+        {
+            object stringArray = new string[] { "A", "B", "C", "D" };
+
+            Console.WriteLine("Testing IEnumerable<T> on array...");
+            string result = String.Empty;
+            foreach (var s in (System.Collections.Generic.IEnumerable<string>)stringArray)
+                result += s;
+
+            if (result != "ABCD")
+            {
+                Console.WriteLine("Failed.");
+                return Fail;
+            }
+        }
+
```

Comment:
Consider adding a test for array interface variance here... This would make sure that the various interfaces and classes have the right variance details specified. For instance...

int[] intArray = new int[]{1,2,3,4,5};
IEnumerable<uint> uintEnum = (uint[])(object)intArray;
uint sum = 0;
foreach (uint x in uintEnum)
{
    sum+=x;
}
if (sum != 15)
{
 return Fail;
}